cdap-ui/app/cdap/components/PipelineConfigurations/Store/ActionCreator.js

```diff
@@ -34,6 +42,52 @@ const applyRuntimeArgs = () => {
   });
 };
 
+// Filter certain preferences from being shown in the run time arguments 
+// They are being represented in other places (like selected compute profile).
+const getFilteredRuntimeArgs = (hideProvided) => {
+  let {runtimeArgs, resolvedMacros} = PipelineConfigurationsStore.getState();
+  let modifiedRuntimeArgs = {};
+  let pairs = [...runtimeArgs.pairs];
+  pairs = pairs
+    .filter(pair => (RUNTIME_ARGS_TO_SKIP_DURING_DISPLAY.indexOf(pair.key) === -1))
+    .map(pair => {
+      if (pair.key in resolvedMacros) {
+        return {
+          notDeletable: true,
+          // This is needed because KeyValuePair will render a checkbox only if the provided is a boolean.
+          provided: hideProvided ? null : pair.provided || false,
+          ...pair
+        };
+      }
+      return {
+        ...pair,
+        provided: hideProvided ? null : pair.provided || false,
```

Comment:
If it's not a macro, I don't think we need the `provided` property?

Reply:
We talked about this in slack right? Marking a property as `provided` is not related to a property being a macro. A user can mark a run time arg as provided if he/she knows that it is going to be provided by a plugin within the pipeline. That was the original usecase for having the provided checkbox.